### PR TITLE
[Messaging] [Warning reporting] Making warning reporting run through the MessageControl plugin

### PR DESCRIPTION
**Pull request by VeithMetro**

### CODE REVIEW

**Comment by VeithMetro** [comment]
@sebaszm I'm not entirely sure what you meant when you wanted to remove `enabled` from the warning reporting's side. As far as I can see, now it is only stored in `WarningReportingControl`, since it inherits from IControl and overrides both of its Enable methods - seems okay?

**Comment by VeithMetro** [comment]
@sebaszm Do we want to go back to `EXTERNAL_HIDDEN`? I don't think it had anything to do with the bounds OOP issue - should I revert it, rebuild and check whether everything still works as intended?

File: Source/core/MessageStore.h [hunk 1]
```diff
@@ -249,6 +278,7 @@ namespace Core {
             virtual bool Default(const Metadata& metadata) const = 0;
             virtual void Push(const Logging& log, const IEvent* message) = 0;
```
Comment (pwielders) [inline Source/core/MessageStore.h]: Thinking out loud and knowing that we are going to get two more streams. I think that for the Push there is no need to explicitly say logging/Tracing or WarningReporting. I guess there is a baseclass for these and on the base class we have a "Length()" and "Serialize()". The implementation might be different but that does not matter (we push a reference) so would it be an opyion to bring this back to 1 method, called:
_virtual void Push(const Metadata& info, const IEvent* message); = 0_ ?

File: Source/messaging/DirectOutput.h [hunk 2]
```diff
@@ -50,6 +50,7 @@ namespace WPEFramework {
             
             void Output(const Core::Messaging::IStore::Logging& log, const Core::Messaging::IEvent* message) const;
```
Comment (pwielders) [inline Source/messaging/DirectOutput.h]: Same as the push, hope we can use a baseclass for this..

**Comment by sebaszm** [comment]
> @sebaszm Do we want to go back to `EXTERNAL_HIDDEN`? I don't think it had anything to do with the bounds OOP issue - should I revert it, rebuild and check whether everything still works as intended?

Yes, use EXTERNAL_HIDDEN.

File: Source/core/MessageStore.cpp [hunk 3]
```diff
@@ -289,6 +335,30 @@ namespace Core {
             return (length);
         }
 
+        string IStore::WarningReporting::ToString(const abbreviate abbreviate) const
+        {
+            string result;
+            const Core::Time now(TimeStamp());
+
+            if (abbreviate == true) {
```
Comment (pwielders) [inline Source/core/MessageStore.cpp]: Maybe better to compare with the enum value in stead of true/false. Would even expect the compiler to generate a warning for this :-)

File: Source/core/MessageStore.cpp [hunk 4]
```diff
@@ -247,6 +267,32 @@ namespace Core {
             return (length);
         }
 
+        string IStore::Tracing::ToString(const abbreviate abbreviate) const
+        {
+            string result;
+            const Core::Time now(TimeStamp());
+
+            if (abbreviate == true) {
```
Comment (pwielders) [inline Source/core/MessageStore.cpp]: Maybe better to compare with the enum value in stead of true/false. Would even expect the compiler to generate a warning for this :-)

File: Source/core/MessageStore.cpp [hunk 5]
```diff
@@ -201,6 +201,26 @@ namespace Core {
             return (length);
         }
 
+        string MessageInfo::ToString(const abbreviate abbreviate) const
+        {
+            string result;
+            const Core::Time now(TimeStamp());
+            string time;
+
+            if (abbreviate == true) {
```
Comment (pwielders) [inline Source/core/MessageStore.cpp]: Maybe better to compare with the enum value in stead of true/false. Would even expect the compiler to generate a warning for this :-)

**Review by pwielders — APPROVED** [review]

**Review by sebaszm — APPROVED** [review]

**Pull request merged**